system/lib/wasmfs/thread_utils.h

```diff
@@ -22,39 +23,73 @@ namespace emscripten {
 // Helper class for synchronously proxying work to a dedicated worker thread,
 // including where the work is asynchronous.
 class ProxyWorker {
-  // The queue we use to proxy work and the dedicated worker.
+public:
+  enum Synchronicity { Sync, Async };
```

Comment:
I've not seen "synchronicity" used in this way before.. is there president for that?   e.g. in JS would I say "what is the synchronicity of that function?  Is it sync or async?"